From 5489e1d22e18740a1924628c5c97096d48dcfbf8 Mon Sep 17 00:00:00 2001
From: Fabio Berton <fabio.berton@ossystems.com.br>
Date: Fri, 15 Feb 2019 10:57:06 -0200
Subject: [PATCH 4/4] use PKG_CHECK_VAR for defining WAYLAND_PROTOCOLS_DATADIR
Organization: O.S. Systems Software LTDA.

This allows to override the wayland-protocols pkgdatadir with the
WAYLAND_PROTOCOLS_DATADIR from environment.

pkgconfig would return an absolute path in
/usr/share/wayland-protocols
for the pkgdatadir value, which is not suitable for cross-compiling.

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
Signed-off-by: Fabio Berton <fabio.berton@ossystems.com.br>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 13fed9daf59..6cff8afb7cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1852,7 +1852,7 @@ for plat in $platforms; do
         if test "x$enable_egl" = xyes; then
           PKG_CHECK_MODULES([WAYLAND_EGL], [wayland-egl-backend >= $WAYLAND_EGL_BACKEND_REQUIRED])
         fi
-        WAYLAND_PROTOCOLS_DATADIR=`$PKG_CONFIG --variable=pkgdatadir wayland-protocols`
+        PKG_CHECK_VAR([WAYLAND_PROTOCOLS_DATADIR], [wayland-protocols >= $WAYLAND_PROTOCOLS_REQUIRED], pkgdatadir)
 
         PKG_CHECK_EXISTS([wayland-scanner >= 1.15],
                           AC_SUBST(SCANNER_ARG, 'private-code'),
-- 
2.21.0

